src/clojure_sandbox/4clojure.clj

```diff
@@ -0,0 +1,605 @@
+           (assoc result v [current]))
+         remaining)))))
+
+(comment (group-by* #(> % 5) #{1 3 6 8}))
+(comment (group-by* #(apply / %) [[1 2] [2 4] [4 6] [3 6]]))
+(comment (group-by* count [[1] [1 2] [3] [1 2 3] [2 3]]))
+
+;; 64 on 4Clojure.
+
+;; 65. Clojure has many collection types, which act in subtly different ways.
+;;     The core functions typically convert them into a uniform "sequence" type
+;;     and work with them that way, but it can be important to understand the
+;;     behavioral and performance differences so that you know which kind is
+;;     appropriate for your application. Write a function which takes a
+;;     collection and returns one of :map, :set, :list, or :vector - describing
+;;     the type of collection it was given. You won't be allowed to inspect
+;;     their class or use the built-in predicates like list? - the point is to
+;;     poke at them and understand their behavior.
+;;
+;; Special Restrictions:
+;; class,type,Class,vector?,sequential?,list?,seq?,map?,set?,instance?,getClass
+(defn type-coll [coll]
+  (let [c (+ 3 (count coll))
+        coll (conj coll [:a :b] [:a :b] [:c :d])
+        nc (count coll)]
+    (cond
+      (= :b (get coll :a)) :map
+      (not= c nc) :set
+      (= [:c :d] (first coll)) :list
+      (= [:c :d] (last coll)) :vector)))
+
+(comment (type-coll {:a 1, :b 2}))
+(comment (type-coll (range (rand-int 20))))
+(comment (type-coll [1 2 3 4 5 6]))
+(comment (type-coll #{10 (rand-int 5)}))
+(comment (map type-coll [{} #{} [] ()]))
+
+;; 66. Given two integers, write a function which returns the greatest common
+;;     divisor.
+;;
+;; https://en.wikipedia.org/wiki/Euclidean_algorithm
+(defn gcd [x y]
+  (loop [a (max x y)
+         b (min x y)]
+    (if (zero? b)
+      a
+      (recur b (mod a b)))))
+
+(comment (gcd 2 4))
+(comment (gcd 10 5))
+(comment (gcd 5 7))
+(comment (gcd 1023 858))
+
+;; 67. Write a function which returns the first x number of prime numbers.
+(defn prime-numbers [x]
+  (letfn [(prime? [n]
+            (cond
+              (<= n 1) false
+              (<= n 3) true
+              :else (loop [[i & remaining] (range 2 (inc (Math/sqrt n)))]
+                      (cond
+                        (nil? i) true
+                        (zero? (rem n i)) false
+                        :else (recur remaining)))))]
+    (take x (filter prime? (range)))))
+
+(comment (prime-numbers 2))
+(comment (prime-numbers 5))
+(comment (last (prime-numbers 100)))
+
+;; 68 on 4Clojure
+
+;; 69. Write a function which takes a function f and a variable number of maps.
+;;     Your function should return a map that consists of the rest of the maps
+;;     conj-ed onto the first. If a key occurs in more than one map, the
+;;     mapping(s) from the latter (left-to-right) should be combined with the
+;;     mapping in the result by calling (f val-in-result val-in-latter)
+(defn merge-with* [f map & maps]
+  (reduce
+   (fn [result map]
```

Comment:
suggestion: rename the `map` to something else here. coz the name conflicts with the function argument

Reply:
Another thing, if your name for the accumulator doesn't have more added context, then its fine just to use `acc`

Reply:
I renamed both `map` locals. They were both in conflict with the `map` function.